app/components/product-form/product-media.tsx

```diff
@@ -33,15 +33,19 @@ export function ProductMedia(props: ProductMediaProps) {
       if (swiperInstance.thumbs) {
         swiperInstance.thumbs.swiper = thumbsSwiper;
         swiperInstance.thumbs.init();
-        swiperInstance.slideTo(activeIndex);
       }
+      swiperInstance.on("slideChange", () => {
+        const realIndex = swiperInstance.realIndex;
+        setActiveIndex(realIndex);
+        thumbsSwiper.slideTo(realIndex);
+      });
```

Comment:
Can we remove this if you set `loop` to `false` ?

Reply:
We should not remove it, try testing with 2 thumbs, it will only slide with the second thumb but cannot slide vice versa.